src/core/unit.c

```diff
@@ -1340,13 +1340,29 @@ const char *unit_description(Unit *u) {
         return strna(u->id);
 }
 
-const char *unit_status_string(Unit *u) {
+const char* unit_status_string(Unit *u, char **combined) {
         assert(u);
+        assert(u->id);
 
-        if (u->manager->status_unit_format == STATUS_UNIT_FORMAT_NAME && u->id)
+        /* Return u->id, u->description, or "{u->id} ({u->description})".
+         * Versions with u->description are only used if it is set.
+         * The last option is used if configured and caller provided 'combined' pointer. */
+
+        if (!u->description ||
+            u->manager->status_unit_format == STATUS_UNIT_FORMAT_NAME ||
+            (u->manager->status_unit_format == STATUS_UNIT_FORMAT_COMBINED && !combined))
                 return u->id;
 
-        return unit_description(u);
+        if (u->description && u->manager->status_unit_format == STATUS_UNIT_FORMAT_COMBINED && combined) {
+                char *t = strjoin(u->id, " (", u->description, ")");
+                if (t) {
+                        *combined = t;
+                        return t;
+                } else
+                        log_oom();
+        }
+
+        return u->description;
```

Comment:
hmm, also, I think we should suppress showing both fields of they are the same anyway (which they will be in many cases, since we initialize Description= from the unit name if we have nothing better)

Reply:
Good point.